include/ginkgo/core/base/executor.hpp

```diff
@@ -1101,6 +1101,66 @@ class executor_deleter<T[]> {
 };
 
 
+/**
+ * This is a deleter that uses an executor's `free` method to deallocate const
+ * data. This is mainly necessary to make Array<const ValueType> work.
+ *
+ * @tparam T  the type of object being deleted
```

Comment:
Perhaps explicitly state, that `T` can't have a `const` qualifier?
```suggestion
 * @tparam T  the type of object being deleted, without const qualifier
```